### PR TITLE
add support for session rotation

### LINKED ISSUE #314: Support rotation of session secrets
I'd like to be able to rotate the session secret without invalidating all active sessions. This should be doable with a change to the environment variable interface and a call to [`SessionHandler::with_older_secrets()`](https://docs.rs/trillium-sessions/latest/trillium_sessions/struct.SessionHandler.html#method.with_older_secrets).

### PR DESCRIPTION
**Pull request by jbr**
closes #314

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1,59 +1,66 @@
...
 use trillium_client::Client;
-use trillium_rustls::RustlsConfig;
-use trillium_tokio::ClientConfig;
 use url::Url;
 
 const POSTMARK_URL: &str = "https://api.postmarkapp.com";
 
 #[derive(Debug, Clone)]
 pub struct Config {
-    pub session_secret: String,
```
Comment (jbr) [inline src/config.rs:14]: these lines changed because I sorted the fields

File: src/config.rs [hunk 2]
```diff
@@ -62,34 +69,39 @@ fn var_optional<T: FromStr>(name: &'static str, default: T) -> Result<T, ConfigE
...
+    Client::new(trillium_testing::connector(ApiMocks::new(
+        POSTMARK_URL,
+        var::<Url>("AUTH_URL").unwrap().as_ref(),
+    )))
 }
 
 impl Config {
     pub fn from_env() -> Result<Self, ConfigError> {
         Ok(Self {
-            database_url: var("DATABASE_URL")?,
```
Comment (jbr) [inline src/config.rs:79]: as above, sorted

File: README.md [hunk 3]
```diff
@@ -30,7 +30,9 @@ An example `.envrc` is provided for optional but recommended use with [`direnv`]
 * `API_URL` -- The public-facing base url for this application
 * `AUTH_CLIENT_ID` -- Auth0-provided client id (see [auth0 config section](#auth0))
 * `AUTH_CLIENT_SECRET` -- Auth0-provided client secret (see [auth0 config section](#auth0))
-* `SESSION_SECRET` -- A cryptographically-randomly secret that is at least 32 bytes long. Future note: trillium sessions support [secret rotation](https://docs.trillium.rs/trillium_sessions/struct.sessionhandler#method.with_older_secrets), but divviup-api does not yet use this
+* `SESSION_SECRETS` -- Comma-joined url-safe-no-pad base64'ed
+  cryptographically-randomly secrets that each at least 32 bytes long
+  after base64. The first one will be used for new sessions.
```
Comment (divergentdave) [inline README.md:35]: nits:
```suggestion
* `SESSION_SECRETS` -- Comma-joined base64url-encoded, without padding,
  cryptographically-randomly secrets that are each at least 32 bytes long
  after base64url decoding. The first one will be used for new sessions.
```

File: src/config.rs [hunk 4]
```diff
@@ -112,3 +124,48 @@ impl AsRef<Client> for Config {
         &self.client
     }
 }
+
+#[non_exhaustive]
+#[derive(thiserror::Error, Debug, Clone, Copy)]
+pub enum SessionSecretsDecodeError {
+    #[error("session secret must be at least 32 bytes after base64 decode")]
+    TooShort,
+    #[error("session secret must be url-safe no-pad base64")]
```
Comment (divergentdave) [inline src/config.rs:133]: nit: standard nomenclature
```suggestion
    #[error("session secret must be base64url without padding")]
```

**Review by divergentdave — APPROVED** [review]

**Pull request merged**